src/bit-fields/src/bit.rs

```diff
@@ -0,0 +1,465 @@
+                assert_eq!(BitMut::<_,2>(&mut x).is_off(),false);
+                ```
+            ")]
+            #[must_use]
+            #[inline]
+            pub fn is_off(&self) -> bool {
+                !bool::from(self)
+            }
+
+            // These values are only evaluated at compile-time, thus a failure can only occur at
+            // compile-time and would be immediately obvious. Thus it is safe to use arithmetic
+            // here.
+            #[doc = concat!("
+                Flips the value of the bit.
+                ```
+                use bit_fields::BitMut;
+                let mut x = 5", stringify!($x), ";
+                BitMut::<_,0>(&mut x).flip();
+                assert_eq!(x,4);
+                BitMut::<_,1>(&mut x).flip();
+                assert_eq!(x,6);
+                BitMut::<_,2>(&mut x).flip();
+                assert_eq!(x,2);
+                ```
+            ")]
+            #[allow(clippy::integer_arithmetic, clippy::arithmetic)]
+            #[inline]
+            pub fn flip(&mut self) {
+                *(self.0) ^= 1 << P;
+            }
+
+            #[doc = concat!("
+                Sets the bit.
+                ```
+                use bit_fields::BitMut;
+                let mut x = 5", stringify!($x), ";
+                BitMut::<_,0>(&mut x).set(false);
+                assert_eq!(x,4);
+                BitMut::<_,1>(&mut x).set(true);
+                assert_eq!(x,6);
+                BitMut::<_,2>(&mut x).set(false);
+                assert_eq!(x,2);
+                ```
+            ")]
+            #[inline]
+            pub fn set(&mut self, x: bool) {
+                if x {
+                    self.on()
+                } else {
+                    self.off()
+                }
+            }
+        }
+    };
+}
+
+bit_mut!(u128);
+bit_mut!(u64);
+bit_mut!(u32);
+bit_mut!(u16);
+bit_mut!(u8);
+
+// impl From<..> for Bit<..>
+macro_rules! bit_from {
+    ($x:ty) => {
+        // These values are only evaluated at compile-time, thus a failure can only occur at
+        // compile-time and would be immediately obvious. Thus it is safe to use arithmetic here.
+        #[allow(clippy::integer_arithmetic, clippy::as_conversions, clippy::arithmetic)]
+        impl<const P: u8> From<&Bit<'_, $x, P>> for bool {
+            #[doc = concat!("
+                ```
+                use bit_fields::Bit;
+                let x = 5", stringify!($x), ";
+                let bit = Bit::<_,0>(&x);
+                assert_eq!(bool::from(&bit),true);
+                let bit = Bit::<_,1>(&x);
+                assert_eq!(bool::from(&bit),false);
+                let bit = Bit::<_,2>(&x);
+                assert_eq!(bool::from(&bit),true);
+                ```
+            ")]
+            #[inline]
+            fn from(this: &Bit<'_, $x, P>) -> Self {
+                (*this.0 >> P) & (1 as $x).to_le() == (1 as $x).to_le()
```

Comment:
nit: I find the following more readable
```suggestion
                *this.0 & MASK != 0
```

Reply:
```rust
(*this.0 & Bit::<$x, P>::MASK) != 0
```